### PR TITLE
Update diagnosis tests

### PR DESCRIPTION
**Pull request by AnilSorathiya**
# Pull Request Description
Tests:
- WeakspotsDiagnosis
- OverfitDiagnosis

These tests takes all the listed all the feature columns for for plots and data. However, if columns are not numerical or categorical types, it doesn't produce good plots.

## What and why?
To produce relevant output

## How to test
Run quick start documentation notebook demo notebook. you can use your dataset to test further.
## Checklist
- [x] What and why
- [ ] Screenshots or videos (Frontend)
- [x] How to test
- [ ] What needs special review
- [ ] Dependencies, breaking changes, and deployment notes
- [x] Labels applied
- [x] PR linked to Shortcut
- [ ] Unit tests added (Backend)
- [x] Tested locally
- [ ] Documentation updated (if required)
- [ ] Environment variable additions/changes documented (if required)

### CODE REVIEW

File: validmind/tests/model_validation/sklearn/OverfitDiagnosis.py [hunk 1]
```diff
@@ -220,6 +220,20 @@ def OverfitDiagnosis(
...
     - Assumes that the binning of features adequately represents the data segments.
     """
+
+    numeric_and_categorical_columns = (
+        datasets[0].feature_columns_numeric + datasets[0].feature_columns_categorical
+    )
+
+    feature_columns = [
+        col for col in datasets[0].feature_columns if col in numeric_and_categorical_columns
+    ]
```
Comment (cachafla) [inline validmind/tests/model_validation/sklearn/OverfitDiagnosis.py:230]: What is this achieving? It seems like we're doing an extra step:

```python
ds_feat_columns = ['num_1', 'num_2', 'num_3', 'cat_1', 'cat_2', 'cat_3', 'obj_1', 'obj_2', 'obj_3']

numeric = ["num_1", "num_2", "num_3"]
cat = ["cat_1", "cat_2", "cat_3"]
numeric_and_categorical_columns = num + cat

feature_columns = [col for col in ds_feat_columns if col in numeric_and_categorical_columns]
>>> feature_columns
['num_1', 'num_2', 'num_3', 'cat_1', 'cat_2', 'cat_3']
```

Comment (AnilSorathiya) [reply validmind/tests/model_validation/sklearn/OverfitDiagnosis.py]: Thanks! good catch.

**Review by juanmleng — APPROVED** [review]
lgtm
